twister: add support for misc emulator (#59414)
zephyrproject-rtos/zephyr: doc/develop/test/twister.rst

sample.yaml).

Running tests on custom emulator
********************************

> Apart from the already supported QEMU and other simulated environments, twister

Reviewer: ```suggestion
Apart from the already supported QEMU and other simulated environments, Twister
```

> supports running any out-of-tree custom & proprietary emulator defined in the

Reviewer: This seems to imply emulators *have* to be proprietary 

👇

```suggestion
supports running any out-of-tree custom emulator defined in the
```

> board's :file:`board.cmake`. To use this type of simulation:
> 
> In :file:`custom_board/custom_board.yaml`, add the following properties:

Reviewer: ```suggestion
board's :file:`board.cmake`. To use this type of simulation, add the following properties to
:file:`custom_board/custom_board.yaml`:
```

::

   simulation: custom
   simulation_exec: <name_of_emu_binary>

> This tells the twister that the board is using a custom emulator called ``<name_of_emu_binary>``,

Reviewer: ```suggestion
This tells Twister that the board is using a custom emulator called ``<name_of_emu_binary>``,
```

make sure this binary exists in the PATH.

> Then, in the :file:`custom_board/board.cmake`, set the emu platforms to ``custom``:

Reviewer: ```suggestion
Then, in :file:`custom_board/board.cmake`, set the supported emulation platforms to ``custom``:
```

::

   set(SUPPORTED_EMU_PLATFORMS custom)

> Finally, implement the custom ``'run'`` target in :file:`custom_board/board.cmake`,
> it should look something like this:

Reviewer: ```suggestion
Finally, implement the ``run_custom`` target in :file:`custom_board/board.cmake`.
It should look something like this:
```

::

   add_custom_target(run_custom
     COMMAND
     <name_of_emu_binary to invoke during 'run'>

>      <any args to be passed into the command, i.e. ${BOARD}, ${APPLICATION_BINARY_DIR}/zephyr/zephyr.elf>

Reviewer: ```suggestion
     <any args to be passed to the command, i.e. ${BOARD}, ${APPLICATION_BINARY_DIR}/zephyr/zephyr.elf>
```
